File: ruby/lib/steam/community/l4d/l4d_explosive.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2009-2011, Sebastian Staudt

require 'steam/community/game_weapon'

# This class represents the statistics of a single explosive weapon for a user
# in Left4Dead
#
# @author Sebastian Staudt
class L4DExplosive

  include GameWeapon

  # Creates a new instance of an explosivve based on the given XML data
  #
  # @param [REXML::Element] weapon_data The XML data of this explosive
  def initialize(weapon_data)
    super weapon_data

    @id    = weapon_data.name
    @shots = weapon_data.elements['thrown'].text.to_i
  end

  # Returns the average number of killed zombies for one shot of this explosive
  #
  # @return [Float] The average number of kills per shot
  def avg_kills_per_shot
    1 / avg_shots_per_kill
  end

end

File: php/lib/steam/community/l4d/L4DExplosive.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2009, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Steam Community
 */

require_once STEAM_CONDENSER_PATH . 'steam/community/GameWeapon.php';

/**
 * @package Steam Condenser (PHP)
 * @subpackage Steam Community
 */
class L4DExplosive extends GameWeapon {

    /**
     * Creates a new instance of L4DExplosive based on the assigned XML data
     * @param $weaponData
     */
    public function __construct($weaponData) {
        parent::__construct($weaponData);

        $this->id    = $weaponData->getName();
        $this->shots = (int) $weaponData->thrown;
    }

    /**
     * Returns the average number of kills for one shot of this explosive.
     * @return float
     */
    public function getAvgKillsPerShot() {
        return 1 / $this->getAvgShotsPerKill();
    }
}
?>
